### ISSUE TITLE
<p> tags in formatted messages cause loads of whitespace [regression]

### ISSUE THREAD

**Issue by eeeeeta** [vector-im/riot-android#2275]
Noticed this regression in the latest dev builds - formatted messages, such as the following example `m.room.message` event, cause there to be lots of annoying whitespace between messages. This didn't happen before, and seems to have been introduced as a result of the reply formatting work.

Example event:
```json
{
  "msgtype": "m.text",
  "body": "test",
  "format": "org.matrix.custom.html",
  "formatted_body": "<p>test</p>"
}
```

Result when sending a lot of them to a room:

![](https://matrix.theta.eu.org/_matrix/media/v1/download/theta.eu.org/eDLNLiWAdVtgakGwtxmqvvBn)

**Comment by bmarty** [comment]
Investigating a few hours on this issue. Not sure if its a regression. So far, I've fixed a '<p>' tag removal issue in the class `VectorMarkdownParser`.

Issue on rendering is due to how `Html.fromHtml()` works. It add lots of space on `<p>` blocks. If someone got an idea how to reduce this space, I take it.

**Comment by ganfra** [comment]
Starting from Android 7.0, you can use flags to manage lines break : FROM_HTML_MODE_COMPACT

**Comment by Matrixcoffee** [comment]
Maybe it's possible to feed the renderer a style sheet somehow.

**Comment by bmarty** [comment]
@ganfra yes, I've seen this, and tried it, but it was not working very well (I do not remember what was wrong)

@Matrixcoffee I've searched for such a solution but not found anything so far

**Comment by bmarty** [comment]
To check in next sprint

**Comment by bmarty** [comment]
@eeeeeta would you please check on the last version (0.8.16) if you still get the issue?

**Comment by eeeeeta** [comment]
@bmarty Looks like it is fixed in 0.8.16 (I wasn't able to test but &Adam and Mathijs confirmed that it works). I think we're good to close this issue!
